pkg/deployment/config.go

```diff
@@ -38,16 +39,36 @@ const (
 	// ProgressDeadlineSeconds. This does not match the K8s default value of 600s.
 	ProgressDeadlineDefault = 120 * time.Second
 
-	registriesSkippingTagResolvingKey = "registriesSkippingTagResolving"
-
 	// ProgressDeadlineKey is the key to configure deployment progress deadline.
 	ProgressDeadlineKey = "progressDeadline"
+
+	// registriesSkippingTagResolvingKey is the config map key for the set of registries
+	// (e.g. ko.local) where tags should not be resolved to digests.
+	registriesSkippingTagResolvingKey = "registriesSkippingTagResolving"
+
+	// queueSidecar resource request keys.
+	queueSidecarCPURequestKey              = "queueSidecarCPURequest"
+	queueSidecarMemoryRequestKey           = "queueSidecarMemoryRequest"
+	queueSidecarEphemeralStorageRequestKey = "queueSidecarEphemeralStorageRequest"
+
+	// queueSidecar resource limit keys.
+	queueSidecarCPULimitKey              = "queueSidecarCPULimit"
+	queueSidecarMemoryLimitKey           = "queueSidecarMemoryLimit"
+	queueSidecarEphemeralStorageLimitKey = "queueSidecarEphemeralStorageLimit"
+)
+
+var (
+	// QueueSidecarCPURequestDefault is the default request.cpu to set for the
+	// queue sidecar. It is set at 25m for backwards-compatibility since this was
+	// the historic default before the field was operator-settable.
+	QueueSidecarCPURequestDefault = resource.MustParse("25m")
 )
 
 func defaultConfig() *Config {
 	return &Config{
 		ProgressDeadline:               ProgressDeadlineDefault,
 		RegistriesSkippingTagResolving: sets.NewString("ko.local", "dev.local"),
+		QueueSidecarCPURequest:         &QueueSidecarCPURequestDefault,
```

Comment:
The CM example doc explicitly says that all fields except this one (to avoid changing current defaults) default to unset (ie no limit). This is same behaviour - again other than cpu to avoid changing current default - as defaults.yaml has for user container resource defaults.

Reply:
But I thought for QP we _do_ set them, vs what we do for UserContainer.

Reply:
yah I think eventually it'd make sense to, but tbh I don't know what the correct values to default to would be for all workloads yet so I limited the initial feature proposal to letting operators set, but maintaining existing behaviour (i.e unset/inherit/infinite) if they don't. Hopefully we can experiment a bit and determine good defaults once this lands.

Reply:
SG